static/js/jira-server-url.js

```diff
@@ -55,14 +56,61 @@ const setErrorMessage = error => {
 	$("#gheServerURL").addClass("has-error");
 };
 
-/**
- * Hides the error messages
- */
 const hideErrorMessage = () => {
 	$("#gheServerURLError").hide();
 	$("#gheServerURL").removeClass("has-error");
 };
 
+const activeRequest = () => {
+	$("#gheServerBtnText").hide();
+	$("#gheServerBtnSpinner").show();
+};
+
+const requestFailed = () => {
+	$("#gheServerBtnText").show();
+	$("#gheServerBtnSpinner").hide();
+};
+
+const checkForOrCreateGitHubApp = (gheServerURL) => {
+	$.ajax({
+		type: "POST",
+		url: "/jira/app-creation",
+		data: {
+			gheServerURL
+		},
+		success: function(data) {
+			const pagePath =  data.moduleKey;
+			AP.navigator.go(
+				"addonmodule",
+				{
+					moduleKey: pagePath
+				}
+			);
+		},
+		error: function(err) {
+			console.error(`Failed to retrieve GH app data. ${err}`)
+			// TODO - build and render error component
+		}
+	});
+}
+
+const verifyGitHubServerUrl = (gheServerURL) => {
+	$.ajax({
+		type: "POST",
+		url: "/jira/verify-server-url",
+		data: {
+			gheServerURL
+		},
+		success: function() {
+			checkForOrCreateGitHubApp(gheServerURL);
```

Comment:
Alternatively we could merge the backend of `verify-server-url` and `checkForOrCreateGitHubApp` in one route, aren't they all doing something related to the GHE url?

----

But doesn't have to be changed per above. Just 2 cents.